CMakeLists.txt

```diff
@@ -81,10 +81,13 @@ set(_alpaka_INCLUDE_DIRECTORY "${_alpaka_ROOT_DIR}/include")
 set(_alpaka_SUFFIXED_INCLUDE_DIR "${_alpaka_INCLUDE_DIRECTORY}/alpaka")
 
 # the sequential accelerator is required for the tests and examples
-if(NOT DEFINED alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE AND (alpaka_BUILD_EXAMPLES OR BUILD_TESTING))
-  option(alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE "enable alpaka serial accelerator" ON)
+if(alpaka_BUILD_EXAMPLES OR BUILD_TESTING)
+  if (NOT DEFINED alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE)
+    option(alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE "enable alpaka serial accelerator" ON)
+  elseif(NOT alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE)
+    message(FATAL_ERROR "Serial backend must be enabled for tests and examples: delete CMakeCache.txt and rerun CMake command")
+  endif()
```

Comment:
If the serial backend is truly necessary for tests and examples, then we should just turn it on and tell the user instead of fatally erroring. Something like (please check if it works):
```cmake
if((NOT alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE  AND (alpaka_BUILD_EXAMPLES OR BUILD_TESTING))
  option(alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE "enable alpaka serial accelerator" ON)
  message(INFO "Turned on serial backend which is required to build examples or tests")
endif()
```
But I would like to have opinions from other maintainers, @psychocoderHPC or @j-stephan.

Reply:
The problem here is that once an option is set (i.e. in the first run of cmake with different options) it cannot be overridden:
```
If <variable> is already set as a normal or cache variable, then the command does nothing (see policy [CMP0077](https://cmake.org/cmake/help/latest/policy/CMP0077.html#policy:CMP0077)).
```
https://cmake.org/cmake/help/latest/command/option.html
so it doesn't complain about the line, but it doesn't change the value.